### ISSUE TITLE
Can't index Dash Testnet/Mainnet for block 1117834 (testnet) / 2159351 (mainnet)

### ISSUE THREAD

**Issue by farukterzioglu** [dgarage/NBXplorer#481]
When NBXplorer tries to get block [1117834](https://blockexplorer.one/dash/testnet/blockHash/000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830) for Dash testnet, it disconnects from the node. If I start from one block earlier, it gets that block but fails at this block again. If I start it after that specific block (e.g. [1117835](https://blockexplorer.one/dash/testnet/blockHash/00000099c75c7de00ffc3f76844b05a3653b92618e139dbf20f9496e04052d77)) it works fine. 

I tried with the latest version from master. It is reproducible. One can try to run NBXplorer with following arguments to force it start from that block. 

Here is the logs from NBXplorer;
(I started from 1 block earlier, block: 1117833, hash: `0000001403d9eeb06344e1f8d10b06e5eb89a03d13efe58c5221c35987d5636d`)
```
info: NBXplorer.Indexer.DASH: Handshaked
info: NBXplorer.Indexer.DASH: Testing RPC connection to ****
info: NBXplorer.Indexer.DASH: RPC connection successful
info: NBXplorer.Indexer.DASH: Full node version detected: 210101
info: NBXplorer.Indexer.DASH: Has txindex support
info: NBXplorer.Indexer.DASH: NBXplorer is correctly whitelisted by the node
info: NBXplorer.Events: DASH: Node state changed: NotStarted => NBXplorerSynching
info: NBXplorer.Indexer.DASH: Current Index Progress not found, start syncing at height 1117833
info: NBXplorer.Indexer.DASH: Node disconnected (Unexpected exception while connecting to socket)
info: NBXplorer.Events: DASH: Node state changed: NBXplorerSynching => NotStarted
info: NBXplorer.Events: DASH: New block 0000001403d9eeb06344e1f8d10b06e5eb89a03d13efe58c5221c35987d5636d (1117833)
fail: NBXplorer.Indexer.DASH: Unhandled exception in the indexer, retrying in 0 seconds
System.OperationCanceledException: The peer has been disconnected
   at NBXplorer.Backend.Indexer.AskNextHeaders(Node node, CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 382
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 215
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 221
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 148
   at NBXplorer.Backend.Indexer.IndexerLoop() in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 95
info: NBXplorer.Indexer.DASH: TCP Connection succeed, handshaking...
info: NBXplorer.Indexer.DASH: Handshaked
info: NBXplorer.Indexer.DASH: Testing RPC connection to ****
info: NBXplorer.Indexer.DASH: RPC connection successful
info: NBXplorer.Indexer.DASH: Full node version detected: 210101
info: NBXplorer.Indexer.DASH: Has txindex support
info: NBXplorer.Indexer.DASH: NBXplorer is correctly whitelisted by the node
info: NBXplorer.Events: DASH: Node state changed: NotStarted => NBXplorerSynching
info: NBXplorer.Indexer.DASH: Node disconnected (Unexpected exception while connecting to socket)
info: NBXplorer.Events: DASH: Node state changed: NBXplorerSynching => NotStarted
fail: NBXplorer.Indexer.DASH: Unhandled exception in the indexer, retrying in 5 seconds
System.OperationCanceledException: The peer has been disconnected
   at NBXplorer.Backend.Indexer.AskNextHeaders(Node node, CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 382
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 215
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 221
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 148
   at NBXplorer.Backend.Indexer.IndexerLoop() in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 95
```

I can see following logs in Dash node;

```
2024-10-24T16:42:14Z Added connection peer=565
2024-10-24T16:42:14Z connection accepted, sock=50, peer=565
2024-10-24T16:42:15Z received: version (115 bytes) peer=565
2024-10-24T16:42:15Z sending version (137 bytes) peer=565
2024-10-24T16:42:15Z send version message: version 70233, blocks=1127539, txrelay=1, peer=565
2024-10-24T16:42:15Z sending sendaddrv2 (0 bytes) peer=565
2024-10-24T16:42:15Z sending verack (0 bytes) peer=565
2024-10-24T16:42:15Z receive version message: NBXplorer-4236754535682635815: version 70223, blocks=0, us=****:19999, txrelay=1, peer=565
2024-10-24T16:42:15Z received: sendaddrv2 (0 bytes) peer=565
2024-10-24T16:42:15Z received: verack (0 bytes) peer=565
2024-10-24T16:42:15Z sending sendheaders (0 bytes) peer=565
2024-10-24T16:42:15Z sending sendcmpct (9 bytes) peer=565
2024-10-24T16:42:15Z sending senddsq (1 bytes) peer=565
2024-10-24T16:42:15Z sending ping (8 bytes) peer=565
2024-10-24T16:42:15Z received: ping (8 bytes) peer=565
2024-10-24T16:42:15Z sending pong (8 bytes) peer=565
2024-10-24T16:42:15Z received: sendheaders (0 bytes) peer=565
2024-10-24T16:42:15Z received: pong (8 bytes) peer=565
2024-10-24T16:42:16Z received: getheaders (69 bytes) peer=565
2024-10-24T16:42:16Z getheaders 1117833 to end from peer=565
2024-10-24T16:42:16Z sending headers (162003 bytes) peer=565
2024-10-24T16:42:16Z SocketHandlerConnected -- remove mapSendableNodes, peer=565
2024-10-24T16:42:16Z received: getdata (361 bytes) peer=565
2024-10-24T16:42:16Z received getdata (10 invsz) peer=565
2024-10-24T16:42:16Z received getdata for: block 0000001403d9eeb06344e1f8d10b06e5eb89a03d13efe58c5221c35987d5636d peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (1614 bytes) peer=565
2024-10-24T16:42:16Z sending block (1649 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (426 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z sending block (392 bytes) peer=565
2024-10-24T16:42:16Z socket closed for peer=565
2024-10-24T16:42:16Z disconnecting peer=565
2024-10-24T16:42:16Z ThreadSocketHandler -- removing node: peer=565 nRefCount=2 fInbound=1 m_masternode_connection=0 m_masternode_iqr_connection=0
2024-10-24T16:42:16Z Cleared nodestate for peer=565
```

When I start from the exact block (`000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830`) that fails;
NBXplorer;
```
....
info: NBXplorer.Indexer.DASH: Current Index Progress not found, start syncing at height 1117834
info: NBXplorer.Indexer.DASH: Node disconnected (Unexpected exception while connecting to socket)
info: NBXplorer.Events: DASH: Node state changed: NBXplorerSynching => NotStarted
fail: NBXplorer.Indexer.DASH: Unhandled exception in the indexer, retrying in 0 seconds
System.OperationCanceledException: The peer has been disconnected
   at NBXplorer.Backend.Indexer.AskNextHeaders(Node node, CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 382
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 215
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 221
   at NBXplorer.Backend.Indexer.IndexerLoopCore(CancellationToken token) in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 148
   at NBXplorer.Backend.Indexer.IndexerLoop() in /Users/***/code/nbxplorer-opensource/NBXplorer/Backend/Indexer.cs:line 95
```
Node;
```
....
2024-10-24T16:58:19Z received getdata for: block 000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830 peer=642
2024-10-24T16:58:19Z sending block (1614 bytes) peer=642
2024-10-24T16:58:19Z sending block (1649 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (426 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z sending block (392 bytes) peer=642
2024-10-24T16:58:19Z socket closed for peer=642
2024-10-24T16:58:19Z disconnecting peer=642
2024-10-24T16:58:19Z ThreadSocketHandler -- removing node: peer=642 nRefCount=2 fInbound=1 m_masternode_connection=0 m_masternode_iqr_connection=0
2024-10-24T16:58:19Z Cleared nodestate for peer=642
```

To run the NBXplorer with that block;
```
"args": [ "--dashstartheight=1117834", "--dashrescan" ], 
```

Config for Dash node;
```
    disablewallet=1
    printtoconsole=1
    rpcuser=****
    rpcpassword=****

    testnet=1

    [test]
    rpcbind=0.0.0.0:8332
    rpcallowip=0.0.0.0/0
    whitelist=*****
    bantime=1

    dbcache=6000
    peertimeout=600
    maxconnections=250

    debug=net
```
Node [version](https://github.com/dashpay/dash/releases/tag/v21.1.1) is latest. 

Same issue is exist with mainnet at block `2159351`.

Multiple NBXplorers have the same issue with different nodes. 
The issue happened first with Dash node 20.x.x, we upgraded the node but it continued. 

Rpc calls to that block return normal;
```
[
  {"jsonrpc":"1.0","id":"block1","method":"getblock","params":["000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830", true]},
  {"jsonrpc":"1.0","id":"block2","method":"getblock","params":["00000099c75c7de00ffc3f76844b05a3653b92618e139dbf20f9496e04052d77", true]}
]
```
```
[
    {
        "result": {
            "hash": "000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830",
            "confirmations": 9731,
            "height": 1117834,
            "version": 536870912,
            "versionHex": "20000000",
            "merkleroot": "b5d7795fdba760540f4718001245c046ab0729dc252f344b00c94a82730a534a",
            "time": 1728446643,
            "mediantime": 1728445675,
            "nonce": 334501,
            "bits": "1e0662c4",
            "difficulty": 0.0006116991487455198,
            "chainwork": "000000000000000000000000000000000000000000000000031c4003ac5c0cca",
            "nTx": 5,
            "previousblockhash": "0000001403d9eeb06344e1f8d10b06e5eb89a03d13efe58c5221c35987d5636d",
            "nextblockhash": "00000099c75c7de00ffc3f76844b05a3653b92618e139dbf20f9496e04052d77",
            "chainlock": true,
            "size": 1614,
            "tx": [
                "c1ea006915c514d779ce377c93283b08eb9dafcfad429e5b8b2e7d84b04a0071",
                "4d157d479aaab0422060194625f65434296d8212b46c1277e38f104e4d3cbd87",
                "6728bb41dbc2dfa447700cf6f06df7f9a73624c45725b971f65f5eb99b93901d",
                "3f864d73d72eed38365932619f97195d95fb1042b6454a68ee0c0f61d959590f",
                "6f0151cae1d5e65305fe2620a4f387108b054874a4ed9a23d4b004d5b08fbf79"
            ],
            "cbTx": {
                "version": 3,
                "height": 1117834,
                "merkleRootMNList": "73a97ce88920e0a491ddb45e72ee412bb03dfb4c644028a9544311c4c7f697ec",
                "merkleRootQuorums": "2c87cbea545a1ebc2d86ebcda10d9547c655b734452feef3f4145722afab5346",
                "bestCLHeightDiff": 0,
                "bestCLSignature": "a8cf7096926c61cbca71a11b150e951ba616991a13f673d0a2a02cfaa753b2cf833b4632ec4f2e723ba3105e144b119a0adabaf96608f6bd535996ce246386e24320d4c1608d62f3ad8767f5aee90caaf82f08bb7d18c9bdb1a956a482c4d6a2",
                "creditPoolBalance": 43704.89651918
            }
        },
        "error": null,
        "id": "block1"
    },
    {
        "result": {
            "hash": "00000099c75c7de00ffc3f76844b05a3653b92618e139dbf20f9496e04052d77",
            "confirmations": 9730,
            "height": 1117835,
            "version": 536870912,
            "versionHex": "20000000",
            "merkleroot": "3adf65811aee537d7a2a3aca5bcdd751b8ea9687150ec4aa626d7b313d02296f",
            "time": 1728446767,
            "mediantime": 1728445819,
            "nonce": 695780,
            "bits": "1e010915",
            "difficulty": 0.003772359584297314,
            "chainwork": "000000000000000000000000000000000000000000000000031c4003ad534772",
            "nTx": 5,
            "previousblockhash": "000001f02c1623e0bb12b54ac505cefdfca3f0f664bf333fc73ae5eafe34b830",
            "nextblockhash": "0000092cf0f7e2e854b24ca63d8565e2732872c5d6ef759fc4a715a9b5c845ac",
            "chainlock": true,
            "size": 1649,
            "tx": [
                "4bdc2cd36cd8f142b06f30a149d6b33ba67b67b9a163bc3562ef10f4dcc34e29",
                "89aa851502c718e527cc69e7933b0d3099393b25646611b1271967e09e31ee6c",
                "df51627d032e00d015873bbc4b5b9cd3eef7a99b86cab272a88dec602fb95d95",
                "2221b6d0eb635de5948b24c5c3982c9dd396fc7c04cf3455cad031ba8f855622",
                "68bf329935f230506c4a0f0b13fdfd11a76b38f9cc4bec89591b675fbe002071"
            ],
            "cbTx": {
                "version": 3,
                "height": 1117835,
                "merkleRootMNList": "73a97ce88920e0a491ddb45e72ee412bb03dfb4c644028a9544311c4c7f697ec",
                "merkleRootQuorums": "fd654eb5fa16a5740c3fa39af7b3e4de2299ed08a19f52aff9f681286576a654",
                "bestCLHeightDiff": 0,
                "bestCLSignature": "a91ea63d9c7daec76f3b0aac3a7541c4b885fd6e0a0a0125f85d30a7af78d9462ae86506400503e34680eb8b0ba91a480bd002e384e1046aef11162bcfdc65c7f8acc1982c0db6bb90ece9d23ce3d990590b31a6b76c6347f32d786d2861fa2d",
                "creditPoolBalance": 43705.67317590
            }
        },
        "error": null,
        "id": "block2"
    }
]
```

**Comment by NicolasDorier** [comment]
@farukterzioglu I'm interested to fix that, but I don't really have any shitcoin node around. 
It seems that the dash node is disconnecting us, but the log doesn't give any idea why.

Don't hesitate to DM me on twitter if I'm late to reply here.

**Comment by NicolasDorier** [comment]
Fixed on NBX 2.5.8, was a NBitcoin bg
